utils/pattern/src/frontend/mod.rs

```diff
@@ -191,6 +192,18 @@ where
     }
 }
 
+impl<B> Pattern<B, B::Store>
+where
+    B: PatternBackend,
+{
+    pub fn from_borrowed_store_unchecked(store: &B::Store) -> Self {
+        Self {
+            _backend: PhantomData,
+            store: store.deref(),
+        }
+    }
+}
```

Comment:
```
error[E0277]: the size for values of type `<B as common::PatternBackend>::Store` cannot be known at compilation time
   --> utils/pattern/src/frontend/mod.rs:199:63
    |
199 |     pub fn from_borrowed_store_unchecked(store: &B::Store) -> Self {
    |                                                               ^^^^ doesn't have a size known at compile-time
    |
    = help: within `frontend::Pattern<B, <B as common::PatternBackend>::Store>`, the trait `Sized` is not implemented for `<B as common::PatternBackend>::Store`, which is required by `frontend::Pattern<B, <B as common::PatternBackend>::Store>: Sized`
note: required because it appears within the type `frontend::Pattern<B, <B as common::PatternBackend>::Store>`
   --> utils/pattern/src/frontend/mod.rs:87:12
    |
87  | pub struct Pattern<Backend, Store: ?Sized> {
    |            ^^^^^^^
    = note: the return type of a function must have a statically known size
help: consider further restricting the associated type
    |
199 |     pub fn from_borrowed_store_unchecked(store: &B::Store) -> Self where <B as common::PatternBackend>::Store: Sized {
    |                                                                    +++++++++++++++++++++++++++++++++++++++++++++++++

```

Reply:
```
fn validate_store(store: &Self::Store)
B::validate_store(store) needs to succeed
```